app/Models/Camp.php

```diff
@@ -5,23 +5,63 @@
 use Illuminate\Database\Eloquent\Model;
 use App\Facades\Util;
 
-
-class Camp extends Model {
-
+class Camp extends Model
+{
     protected $table = 'camp';
     public $timestamps = false;
+    const AGREEMENT_CAMP = "Agreement";
+
+    /**
+     * The attributes that are mass assignable.
+     *
+     * @var array
+     */
+    protected $fillable = ['topic_num', 'parent_camp_num', 'key_words', 'language', 'note', 'submit_time', 'submitter_nick_id', 'go_live_time', 'title', 'camp_name', 'camp_num'];
+
+    /**
+     * The attributes that should be hidden for arrays.
+     *
+     * @var array
+     */
+    protected $hidden = [];
+
+    public static function getLiveCamp($topicnum, $campnum, $filter)
+    {
+        if ((isset($filter) && $filter == "default")) {
+
+            return self::where('topic_num', $topicnum)
+                ->where('camp_num', '=', $campnum)
+                ->where('objector_nick_id', '=', NULL)
+                ->where('go_live_time', '<=', time())
+                ->latest('submit_time')->first();
+        } else  if ((isset($filter) && $filter == "review")) {
 
+            return self::where('topic_num', $topicnum)
+                ->where('camp_num', '=', $campnum)
+                ->where('objector_nick_id', '=', NULL)
+                ->latest('submit_time')->first();
+        } else if ((isset($filter) && $filter == "bydate")) {
+            $asofdate = strtotime(date('Y-m-d H:i:s', strtotime($_REQUEST['asofdate'])));
 
-    public static function campLink($topicNum,$campNum,$title,$campName){
+            return self::where('topic_num', $topicnum)
+                ->where('camp_num', '=', $campnum)
+                ->where('objector_nick_id', '=', NULL)
+                ->where('go_live_time', '<=', $asofdate)
+                ->latest('submit_time')->first();
+        } else {
+            return self::where('topic_num', $topicnum)
+                ->where('objector_nick_id', '=', NULL)
+                ->latest('submit_time')->first();
+        }
+    }
+
+    public static function campLink($topicNum, $campNum, $title, $campName)
+    {
         $title = preg_replace('/[^A-Za-z0-9\-]/', '-', $title);
```

Comment:
The 4 lines of code is exists (I just noticed) in Topic.php also. Can we make a function ?